crates/debug/src/transform/attr.rs

```diff
@@ -252,6 +256,12 @@ where
                 pending_di_refs.insert(current_scope_id, attr.name(), offset);
                 continue;
             }
+            AttributeValue::DebugAddrBase(offset) => {
+                if offset.0 != 8 {
```

Comment:
This is used in `DebugAddrIndex` and it looks like it is compiled unit attribute?

Reply:
I handle `DebugAddrIndex` in 4 places:
```
crates/debug/src/gc.rs:                } else if let read::AttributeValue::DebugAddrIndex(i) = low_pc {
crates/debug/src/transform/attr.rs:            AttributeValue::Addr(_) | AttributeValue::DebugAddrIndex(_)
crates/debug/src/transform/attr.rs:            AttributeValue::DebugAddrIndex(i) => {
crates/debug/src/transform/range_info_builder.rs:            } else if let Some(AttributeValue::DebugAddrIndex(i)) =
```
Relevant occurrences for `DebugAddrBase`
```
$ git grep -n DebugAddrBase
crates/debug/src/gc.rs:174:                    let a = dwarf.debug_addr.get_address(4, DebugAddrBase(8), i)?;
crates/debug/src/transform/attr.rs:85:                let u = context.debug_addr.get_address(4, DebugAddrBase(8), i)?;
crates/debug/src/transform/attr.rs:259:            AttributeValue::DebugAddrBase(offset) => {
crates/debug/src/transform/attr.rs:261:                    bail!("Unexpected (8) DebugAddrBase: {}", offset.0)
crates/debug/src/transform/mod.rs:43:    debug_addr_base: DebugAddrBase<R::Offset>,
crates/debug/src/transform/mod.rs:63:        debug_addr_base: DebugAddrBase(0),
crates/debug/src/transform/range_info_builder.rs:36:                context.debug_addr.get_address(4, DebugAddrBase(8), i)?
```
attr.rs:259 should probably be done earlier, and just ignored in attr.rs

Possibly passing it to `debug_addr_base:` in mod.rs:63.

I am also puzzled by the `0` vs. `8` thing. (`clang-10` creates `8` when `-gdwarf-5`).

Reply:
One needs a `DebugAddrBase` in the CU, so that its `DebugAddrIndex`s can be resolved to `Addr`s. The module provides the `.debug_addr` section, which is shared between CUs.

Reply:
> The module provides the .debug_addr section, which is shared between CUs.

Correct. That what is why DebugAddrIndex needs CU's DebugAddrBase.

Can debug_addr_base be passed from the clone_unit?

Reply:
Actually `gimli` already fills this in for us:
```
                    constants::DW_AT_addr_base => {
                        if let AttributeValue::DebugAddrBase(base) = attr.value() {
                            unit.addr_base = base;
                        }
```
we can access `unit.addr_base` (and should).

Reply:
Commit cc1e974 ends the `DebugAddrBase` confusion. PTAL.